src/main/scala/com/microsoft/hyperspace/util/PathUtils.scala

```diff
@@ -20,10 +20,11 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{Path, PathFilter}
 
 object PathUtils {
-  def makeAbsolute(path: String): Path = makeAbsolute(new Path(path))
+  def makeAbsolute(path: String, hadoopConfiguration: Configuration = new Configuration): Path =
+    makeAbsolute(new Path(path), hadoopConfiguration)
 
-  def makeAbsolute(path: Path): Path = {
-    val fs = path.getFileSystem(new Configuration)
+  def makeAbsolute(path: Path, hadoopConfiguration: Configuration): Path = {
```

Comment:
Any reason this doesn't have a default parameter whereas the above one does?

Reply:
Yes - **overloading**. The `def makeAbsolute(path: String, hadoopConfiguration: Configuration = new Configuration)` is overloading the one at line 26. Compiler complains when both have default parameters:

```
PathUtils.scala:22: in object PathUtils, multiple overloaded alternatives of 
makeAbsolute define default arguments
```